### PR TITLE
Catch errors thrown by "decodeURIComponent"

### PR DESCRIPTION
**Pull request by konstantin-msft**
- Catch errors thrown by "decodeURIComponent"
- Negate state condition check for better readability.

### CODE REVIEW

**Review by tnorling — APPROVED** [review]
Any tests you can add for the new error cases?

**Comment by konstantin-msft** [comment]
> Any tests you can add for the new error cases?

done

File: lib/msal-common/src/response/ResponseHandler.ts [hunk 1]
```diff
@@ -82,15 +82,33 @@ export class ResponseHandler {
         cryptoObj: ICrypto
     ): void {
         if (!serverResponseHash.state || !cachedState) {
-            throw !serverResponseHash.state
-                ? ClientAuthError.createStateNotFoundError("Server State")
-                : ClientAuthError.createStateNotFoundError("Cached State");
+            throw serverResponseHash.state
```
Comment (lalimasharda) [inline lib/msal-common/src/response/ResponseHandler.ts:85]: Is this change for better readability?

**Review by lalimasharda — APPROVED** [review]

File: lib/msal-common/src/response/ResponseHandler.ts (hunk 1 above)
Comment (konstantin-msft) [reply lib/msal-common/src/response/ResponseHandler.ts:85]: Yep

**Review by peterzenz — APPROVED** [review]

**Pull request merged**